client/src/components/mapping/FhirMappingPanel/FhirResourceTree/nodeLabel.tsx

```diff
@@ -3,25 +3,55 @@ import { Menu, MenuItem, ContextMenu, Tooltip } from '@blueprintjs/core';
 import { Attribute } from '@arkhn/fhir.ts';
 
 import AddExtensionSelect from 'components/selects/addExtensionSelect';
+import AddSliceSelect from 'components/selects/addSliceSelect';
 import { ApolloProvider, useApolloClient } from 'react-apollo';
 
 interface NodeLabelProps {
   attribute: Attribute;
   addNodeCallback: any;
+  addSliceCallback: any;
   addExtensionCallback: any;
   deleteNodeCallback: any;
 }
 
 export const NodeLabel = ({
-  attribute: { types, definition, id, isArray, isItem, isSlice, extensions },
+  attribute: {
+    types,
+    definition,
+    id,
+    isArray,
+    isItem,
+    isSlice,
+    slices,
+    extensions
+  },
   addNodeCallback,
+  addSliceCallback,
   addExtensionCallback,
   deleteNodeCallback
-}: NodeLabelProps) => {
+}: NodeLabelProps): React.ReactElement => {
   const client = useApolloClient();
-  const renderAddItem = () => (
-    <MenuItem icon={'add'} onClick={addNodeCallback} text={'Ajouter un item'} />
-  );
+  const renderAddItem = () => {
```

Comment:
I think it's best to have a function renderAddSlice and you do the conditional rendering in "showContextMenu"